semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/NodeTypeStarters.scala

```diff
@@ -173,6 +173,13 @@ class NodeTypeStarters(cpg: Cpg) {
   def method(fullName: String): Traversal[Method] =
     method.fullName(fullName)
 
+  /**
+    * Shorthand for `cpg.method.fullNameExact(fullName)`, but first looks up the fullname in the index.
+    * */
+  def methodFullNameExact(value: String): Traversal[Method] = {
```

Comment:
I'd go a different route:

`cpg.method` should return an object that knows about indices, instead of a stupid iterator; e.g. it could implement some new trait like `HasFastIndices`.

Then, traversal filters (like e.g. `.fullName` on a traversal of methods) should check whether their parent traversal `isInstanceOf[HasFastIndices]` and if so, ask it whether the relevant key ("FULL_NAME") is indexed.

There is no need to change any API with that approach.

The entire dance is done eagerly and once per traversal construction, just like the regex dance (`.name(string)` checks whether the string is equivalent to non-regex; if so, it uses `nameExact` logic, otherwise it eagerly precompiles the regex and allocates a matcher that can be reused, such that later traversal filtering is allocation-free).